src/pages/workspaces/workspace/JobHistory.js

```diff
@@ -142,157 +144,170 @@ const JobHistory = _.flow(
 
   render() {
     const { namespace, name, ajax: { Workspaces }, workspace: { workspace: { workflowCollectionName } } } = this.props
-    const { submissions, loading, aborting, newSubmissionId, highlightNewSubmission, linkToFC } = this.state
+    const { submissions, loading, aborting, newSubmissionId, highlightNewSubmission, linkToFC, textFilter } = this.state
 
-    return div({ style: styles.submissionsTable }, [
-      submissions && !!submissions.length && h(AutoSizer, [
-        ({ width, height }) => h(FlexTable, {
-          width, height, rowCount: submissions.length,
-          hoverHighlight: true,
-          styleRow: rowIndex => {
-            const { submissionId } = submissions[rowIndex]
-            if (newSubmissionId === submissionId) {
-              return {
-                transition: `background-color ${animationLengthMillis}ms cubic-bezier(0.33, -2, 0.74, 0.05)`,
-                backgroundColor: highlightNewSubmission ? colors.blue[5] : 'white'
+    const filteredSubmissions = _.filter(sub => {
+      const subAsText = JSON.stringify(_.values(sub))
+      return _.every(term => Utils.textMatch(term, subAsText), textFilter.split(/s+/))
+    }, submissions)
+
+    return h(Fragment, [
+      h(SearchInput, {
+        style: { width: 300, margin: '1rem 1rem 0', borderColor: colors.gray[3], alignSelf: 'flex-end' },
+        placeholder: 'Filter',
+        onChange: ({ target: { value } }) => this.setState({ textFilter: value }),
+        value: textFilter
+      }),
+      div({ style: styles.submissionsTable }, [
+        filteredSubmissions && !!filteredSubmissions.length && h(AutoSizer, [
```

Comment:
Is our convention to coerce this into a boolean?
```suggestion
        !!filteredSubmissions && !!filteredSubmissions.length && h(AutoSizer, [
```